### PR TITLE
Restyle header

### PR DESCRIPTION
**Pull request by jreis13**
Why:

- Header changes on the design;

These changes address the need by:

- Updating `Header` and its styles;
- Adding home-page and post images;
- Adding `MainHero`;
- Adding `Layouts`;
- Updating `Logo`;
- Updating `NavLinks`;

### CODE REVIEW

File: src/components/Author/index.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import dateFormat from "dateformat"
+import PropTypes from "prop-types"
+
+import Image from "../../images/team/placeholder-v.jpg"
+
+import * as styles from "./index.module.scss"
+
+const renderAuthor = ({ key, name }) => (
+  <Link to={`/blog/author/${key}`}>{name}</Link>
+)
```
Comment (mirhamasala) [inline src/components/Author/index.jsx:12]: I believe @gabrielpoca converted all the arrow functions

```suggestion
function renderAuthor({ key, name })  {
 return <Link to={`/blog/author/${key}`}>{name}</Link>
}
```

File: src/components/Author/index.module.scss [hunk 2]
```diff
@@ -0,0 +1,18 @@
+.root {
+  display: flex;
+  align-items: center;
+
+  margin-bottom: 28px;
+
+  @media (min-width: 950px) {
+    margin-bottom: 0;
+  }
```
Comment (mirhamasala) [inline src/components/Author/index.module.scss:9]: Let's move out layout properties to a layout component or to the "page component" that's holding the `Author` component

File: src/components/Author/index.module.scss [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+.picture {
```
Comment (mirhamasala) [inline src/components/Author/index.module.scss:12]: Same

File: config/node/createPages.js [hunk 4]
```diff
@@ -71,9 +68,33 @@ const createBlogPostPages = async ({ createPage, graphql }) => {
...
+
+  results.data.allMarkdownRemark.distinct.forEach((tagKey) =>
+    createPage({
+      component,
+      context: { tagKey },
+      path: path.posix.join("/tags/", tagKey),
+    })
+  )
+}
+
```
Comment (mirhamasala) [inline config/node/createPages.js:93]: @gabrielpoca Do we want tag/category pages?

File: src/components/Header/index.jsx [hunk 5]
```diff
@@ -1,43 +1,41 @@
...
+      setDesktop(window.innerWidth > 950)
+    })
+  }, [])
+
+  useEffect(() => {
+    if (window.innerWidth > 950) {
+      setDesktop(true)
+    } else {
+      setDesktop(false)
+    }
```
Comment (mirhamasala) [inline src/components/Header/index.jsx:30]: Let's read up on when effects need to be cleaned up.

File: src/components/Header/index.jsx [hunk 6]
```diff
@@ -1,43 +1,41 @@
...
+import Logo from "../Logo"
+import NavLinks from "../NavLinks"
+import NavMenu from "../NavMenu"
 
 import * as styles from "./index.module.scss"
 
-const BLOG_PATH = "/blog/"
+function Header() {
+  const [scroll, setScroll] = useState(0)
+  const [desktop, setDesktop] = useState(false)
```
Comment (mirhamasala) [inline src/components/Header/index.jsx:11]: It's good practice to prefix Boolean states with `is` or `has` etc.

File: src/components/Header/index.jsx (hunk 5 above)
Comment (mirhamasala) [inline src/components/Header/index.jsx:30]: Can be a ternary.

File: src/components/Header/index.jsx [hunk 7]
```diff
@@ -1,43 +1,41 @@
...
+
+  useEffect(() => {
+    window.addEventListener("scroll", () => {
+      setScroll(window.scrollY > 756)
+    })
+  }, [])
```
Comment (mirhamasala) [inline src/components/Header/index.jsx:17]: Let's look into dependencies arrays for event listeners

File: src/components/Header/index.jsx [hunk 8]
```diff
@@ -1,43 +1,41 @@
...
-            </li>
-            <li className={styles.navLink}>
-              <NavLink to="mailto:contact@subvisual.com" title="Contact" blank>
-                Contact
-              </NavLink>
-            </li>
-          </ul>
-        </nav>
-      </div>
+    <header className={scroll ? styles.scrolled : styles.root}>
```
Comment (mirhamasala) [inline src/components/Header/index.jsx:34]: Doesn't `scroll` always return `true`?

File: src/components/Header/index.module.scss [hunk 9]
```diff
@@ -1,58 +1,60 @@
...
 
-  height: 26px;
-  padding: 0;
+  padding-top: 40px;
+  padding-bottom: 40px;
+  padding-left: 28px;
 
-  margin: 0;
+  background-color: #fcfcfc;
+  color: #403f4c;
```
Comment (mirhamasala) [inline src/components/Header/index.module.scss:39]: Let's use vars here

File: src/components/MainHero/index.module.scss [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  height: 816px;
+  margin-bottom: 120px;
+
+  background-repeat: no-repeat;
+  background-size: cover;
+
+  border-bottom-right-radius: 60px;
+  border-bottom-left-radius: 60px;
+
+  color: #fff;
```
Comment (mirhamasala) [inline src/components/MainHero/index.module.scss:11]: 🎨

File: src/components/NavLinks/index.jsx [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    <nav>
+      <ul className={styles.root}>
+        <li className={styles.link}>
+          <NavLink to={SERVICES_PATH}>Services</NavLink>
+        </li>
+        <li className={styles.link}>
+          <NavLink to={VENTURES_PATH}>Ventures</NavLink>
+        </li>
+        <li className={styles.link}>
+          <NavLink to={PEOPLE_PATH} title="People">
```
Comment (mirhamasala) [inline src/components/NavLinks/index.jsx:24]: Leftover `title`

File: src/components/NavLinks/index.module.scss [hunk 12]
```diff
@@ -0,0 +1,34 @@
+.root {
+  display: flex;
+  align-items: center;
+
+  height: 33px;
+
+  margin-right: 86px;
```
Comment (mirhamasala) [inline src/components/NavLinks/index.module.scss:7]: 🚨

File: src/components/NavMenu/index.module.scss [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+
+  text-align: end;
+}
+
+.link {
+  margin-bottom: 8px;
+  &:last-child {
+    margin-bottom: 0;
+  }
+}
```
Comment (mirhamasala) [inline src/components/NavMenu/index.module.scss:36]: Same .. probably can be a `flex` or `grid` with `gap`

File: src/components/PostTitle/index.jsx [hunk 14]
```diff
@@ -0,0 +1,9 @@
+import React from "react"
+
+import * as styles from "./index.module.scss"
+
+function PostTitle({ children }) {
+  return <p className={styles.root}>{children}</p>
```
Comment (mirhamasala) [inline src/components/PostTitle/index.jsx:6]: Should this not be an `h1`?

File: src/components/PostTitle/index.module.scss [hunk 15]
```diff
@@ -0,0 +1,25 @@
+.root {
+  padding-left: 28px;
+  margin-top: 290px;
```
Comment (mirhamasala) [inline src/components/PostTitle/index.module.scss:3]: 🚨

File: src/components/PostTitle/index.module.scss [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+  font-size: 32px;
+  line-height: 35px;
+
+  @media (min-width: 768px) {
+    font-size: 64px;
+    line-height: 70px;
+  }
+
+  @media (min-width: 950px) {
+    padding-left: 56px;
```
Comment (mirhamasala) [inline src/components/PostTitle/index.module.scss:16]: 🚨

File: src/components/PostTitle/index.module.scss [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+
+    font-size: 84px;
+    line-height: 92.4px;
+  }
+
+  @media (min-width: 1300px) {
+    padding-left: 118px;
```
Comment (mirhamasala) [inline src/components/PostTitle/index.module.scss:23]: 🚨

File: src/components/Tags/index.jsx [hunk 18]
```diff
@@ -0,0 +1,27 @@
+import React from "react"
+
+import * as styles from "./index.module.scss"
+
+function Tags({ tags }) {
+  return (
+    <ul className={styles.tags}>
+      {tags ? (
```
Comment (mirhamasala) [inline src/components/Tags/index.jsx:8]: What happens to this component if there are no `tags`? Should we assume that if we're using this component there are always `tags`?

File: src/components/Tags/index.module.scss [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+  display: flex;
+
+  list-style: none;
+
+  font-size: 12px;
+}
+
+.tag {
+  padding: 6px 12px;
+  margin-right: 16px;
```
Comment (mirhamasala) [inline src/components/Tags/index.module.scss:11]: Same as before. Look into using a `flex` container or `grid` with `gap`

File: src/components/Title/index.jsx [hunk 20]
```diff
@@ -0,0 +1,9 @@
+import React from "react"
+
+import * as styles from "./index.module.scss"
+
+function Title({ children }) {
+  return <p className={styles.root}>{children}</p>
```
Comment (mirhamasala) [inline src/components/Title/index.jsx:6]: `h1`?

File: src/components/Title/index.module.scss [hunk 21]
```diff
@@ -0,0 +1,27 @@
+.root {
+  padding-left: 28px;
+  margin-top: 290px;
```
Comment (mirhamasala) [inline src/components/Title/index.module.scss:3]: 🚨

File: src/pages/tags.js [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+        fieldValue
+        totalCount
+      }
+    }
+  }
+`
+
+function capitalize(string) {
+  return string.charAt(0).toUpperCase() + string.slice(1)
+}
```
Comment (mirhamasala) [inline src/pages/tags.js:29]: CSS: `text-transform: capitalize;`

**Review by mirhamasala — CHANGES_REQUESTED** [review]
Wow, lots of good work here @jreis13! Just needs a bit of clean up but nothing major. Well done! (Let's just check with @gabrielpoca what to do about those tag pages since I don't believe there's a design/plan for it. Though, a really nice idea!)

**Comment by gabrielpoca** [comment]
@mirhamasala @jreis13  I squashed my `gp-show-categories` into a single commit. Just merge this branch into that one and then we can merge everything together to master.

**Comment by mirhamasala** [comment]
@jreis13

**Pull request merged**